amethyst_core/src/hidden.rs

```diff
@@ -14,9 +14,32 @@ impl Component for Hidden {
 
 /// Like [Hidden](struct.Hidden.html), but can propagate through children when the [HideHierarchySystem](struct.HideHierarchySystem.html)
 /// is enabled in the [RenderBundle](struct.RenderBundle.html).
-#[derive(Clone, Debug, Default)]
-pub struct HiddenPropagate;
+#[derive(Clone, Debug)]
+pub struct HiddenPropagate {
+    pub(crate) was_propagated: bool,
+}
 
 impl Component for HiddenPropagate {
-    type Storage = FlaggedStorage<Self, NullStorage<Self>>;
+    type Storage = FlaggedStorage<Self>;
```

Comment:
not: I’d prefer explicit backing storage type, it’s more obvious for new users IMHO